File: nick_derobertis_site/research_page/research_project_pane/research_project_pane_component.py
import param

from nick_derobertis_site.common.component import HTMLComponent
from .research_project_pane_model import ResearchProjectPaneModel


class ResearchProjectPaneComponent(HTMLComponent):
    model = param.ClassSelector(class_=ResearchProjectPaneModel)
    is_reversed: bool = param.Boolean(default=False)

File: frontend/nick-derobertis-site/src/app/research/research-project-pane/research-project-pane.component.ts
import { Component, Input, OnInit } from '@angular/core';
import { ResearchModel } from './research-model';

@Component({
  selector: 'nds-research-project-pane',
  templateUrl: './research-project-pane.component.html',
  styleUrls: ['./research-project-pane.component.scss'],
})
export class ResearchProjectPaneComponent implements OnInit {
  @Input() model: ResearchModel;
  @Input() isReversed: boolean = false;

  constructor() {}

  ngOnInit(): void {}

  get containerClassStr(): string {
    let classStr: string = 'row research-pane';
    if (this.isReversed) {
      classStr += ' flex-row-reverse';
    }
    return classStr;
  }
}
